src/libraries/System.Private.DataContractSerialization/src/System/Runtime/Serialization/DataContract.cs

```diff
@@ -1116,16 +1117,24 @@ internal virtual bool HasRoot
                 set { }
             }
 
-            internal virtual XmlDictionaryString TopLevelElementName
+            internal virtual XmlDictionaryString? TopLevelElementName
```

Comment:
`[DisallowNull]`?

Reply:
How does `[DisallowNull]` work with overrides? This one is a bit tricky because the base class has:

```C#
class DataContract
{
    XmlDictionaryString _name; // can't be null

    public XmlDictionaryString Name { get { } set{ _name = value; } }

    public virtual XmlDictionaryString? TopLevelElementName
    {
        get { return _name; } set{ _name = value; }
    }
}
```

but then a derived class does:

```C#
class DerivedDataContract : DataContract
{
    XmlDictionaryString? _topLevelName ; // can be null

    public override XmlDictionaryString? TopLevelElementName
    {
        get { return _topLevelName ; } set{ _topLevelName = value; }
    }
}
```

And when trying to clone the objects, now we have a problem:

https://github.com/dotnet/runtime/blob/6072e4d3a7a2a1493f514cdf4be75a3d56580e84/src/libraries/System.Private.DataContractSerialization/src/System/Runtime/Serialization/ClassDataContract.cs#L1508-L1509

The getter can be `null`, but the setter doesn't allow for it.